crm_properties.go

```diff
@@ -0,0 +1,66 @@
+package hubspot
+
+import "fmt"
+
+const (
+	crmPropertiesPath = "properties"
+)
+
+// CrmPropertiesService is an interface of CRM crm/v3/properties interface..
+// Reference: https://developers.hubspot.com/docs/api/crm/properties
+type CrmPropertiesService interface {
+	List(string) (map[string]interface{}, error)
+	Create(string, map[string]interface{}) (map[string]interface{}, error)
+	Get(string, string) (map[string]interface{}, error)
+	Delete(string, string) error
+	Update(string, string, map[string]interface{}) (map[string]interface{}, error)
+}
+
+// CrmPropertiesServiceOp handles communication with the CRM properties endpoint.
+type CrmPropertiesServiceOp struct {
+	client            *Client
+	crmPropertiesPath string
+}
+
+var _ CrmPropertiesService = (*CrmPropertiesServiceOp)(nil)
+
+func (s *CrmPropertiesServiceOp) List(objectType string) (map[string]interface{}, error) {
+	path := fmt.Sprintf("%s/%s", s.crmPropertiesPath, objectType)
+	resource := make(map[string]interface{})
+	if err := s.client.Get(path, &resource, nil); err != nil {
+		return nil, err
+	}
+	return resource, nil
+}
+
+func (s *CrmPropertiesServiceOp) Get(objectType, propertyName string) (map[string]interface{}, error) {
+	path := fmt.Sprintf("%s/%s/%s", s.crmPropertiesPath, objectType, propertyName)
+	resource := make(map[string]interface{})
+	if err := s.client.Get(path, &resource, nil); err != nil {
+		return nil, err
+	}
+	return resource, nil
+}
+
+func (s *CrmPropertiesServiceOp) Create(objectType string, reqData map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
If you make the request a map, you won't know which value to send.
Although it will be long, I think it would be better to list.
```go
func (s *CrmPropertiesServiceOp) Create(objectType, propertyName, label string ...)
```

Reply:
Hi,

I'm not sure this is a great idea, there are a ton of attributes on a property and if they add any in the future, the function signature would change for any existing code.

Additionally, I have a use case where I want to be able to define "hubspot migrations" in a folder in my application that will allow me to run them in production and in testing in a controlled and  defined way.

I tried having Create/Update take a `CrmProperty` as the parameter rather than `interface{}`, however this gets really messy as you can accidentally update a boolean property since golang defaults to false...  `omitempty` would  break setting booleans to false as false is empty.   The alternative is to make every field in the structure a pointer which I could also do, but that makes the code less user friendly, (e.g. *bool everywhere)

Contact in the CRM interface also uses `interface{}` for its inputs to create/update so this is consistent behavior...

I defined a CRMPropety type and I've updated the CrmProperties service to this interface so that the result is concrete:

```
type CrmPropertiesService interface {
	List(string) (*CrmPropertiesList, error)
	Create(string, interface{}) (*CrmProperty, error)
	Get(string, string) (*CrmProperty, error)
	Delete(string, string) error
	Update(string, string, interface{}) (*CrmProperty, error)
}
``` 

Thoughts?  I will be making a similar update to the schemas endpoint...

Reply:
> I defined a CRMPropety type and I've updated the CrmProperties service to this interface so that the result is concrete:
Thoughts? I will be making a similar update to the schemas endpoint...

I think the current form is better because it was intended for this type of use!
https://github.com/belong-inc/go-hubspot/blob/main/example_test.go#L266

> I tried having Create/Update take a CrmProperty as the parameter rather than interface{}, however this gets really messy as you can accidentally update a boolean property since golang defaults to false... omitempty would break setting booleans to false as false is empty. The alternative is to make every field in the structure a pointer which I could also do, but that makes the code less user friendly, (e.g. *bool everywhere)

contact solves this problem by using a pointer string. I'm not sure if this is the best solution, but if you have a better idea, please let me know.
https://github.com/belong-inc/go-hubspot/blob/main/contact.go#L28

Reply:
So I'm pretty sure using a `hubspot.Contact` object as the input to an API endpoint would currently have unintended behavior.

Consider:

```
	c := hubspot.Contact{HsEmailQuarantined: true}
	json1, _ := json.MarshalIndent(c, "", " ")
	fmt.Printf("%s\n", json1)

	c2 := hubspot.Contact{HsEmailQuarantined: false}
	json2, _ := json.MarshalIndent(c2, "", " ")
	fmt.Printf("%s\n", json2)
```

The output to this code is 

```
{
 "hs_email_quarantined": true
}
{}
```


If we want to change a value that is currently `true` to `false` and pass it to the API handlers, that field will never be sent to Hubspot.  If we don't use `omitempty` and the value is `nil`, then it gets marshalled into `null`

https://github.com/golang/go/issues/22480 covers a proposal to allow include `omitnil` in `encoding/json` but it doesn't appear to have made it to the language yet.

I'll try to figure out something for this when I have time....

Reply:
On further inspection, it appears that `*bool` and `omitempty` would behave correctly.

So I can use this in my code, however I'd suggest that instances of `HsBool` elsewhere in code, need to be pointers...